apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -362,6 +354,54 @@ public override async Task ModifyAckDeadlineAsync(SubscriptionName subscription,
                     ids.AddRange(ackIds.Select(id => new TimedId(_clock.GetCurrentDateTimeUtc(), id)));
                 }
             }
+
+            /// <summary>
+            /// This method does the following:
+            /// <list type="bullet">
+            /// <item> Checks if subscription under test is exactly once or not.</item>
+            /// <item> If it's not exactly once, it will always throw the specified exception for the specified number of times as <c>_numberOfAckModifyAckDeadlineFailures</c>.</item>
+            /// <item> If it's exactly once, it first checks if the entire request has failed permanently based on <c>StatusCode</c>. If yes, it will set the flag to throw the specified exception.</item>
+            /// <item> If the entire request has not failed permanently, it will check the error metadata to see if it contains permanent error for the given ackId. If yes, it will set the flag to throw the specified exception.</item>
+            /// <item> If the flag to throw the specified exception is set and the specified number of exceptions are not thrown already, the specified exception is thrown, else exception is not thrown.</item>
+            /// </list>
+            /// </summary>
+            /// <param name="ackIds">The list of acknowledgement ids being processed.</param>
+            private void MayBeThrowException(IEnumerable<string> ackIds)
```

Comment:
Nit: MayBe => Maybe (as maybe is one word)

Reply:
Done.